src/main/java/seedu/module/logic/commands/LinkCommand.java

```diff
@@ -0,0 +1,95 @@
+package seedu.module.logic.commands;
+
+import java.util.List;
+import static java.util.Objects.requireNonNull;
+
+import seedu.module.commons.core.Messages;
+import seedu.module.commons.core.index.Index;
+import seedu.module.logic.commands.exceptions.CommandException;
+import seedu.module.model.Model;
+import seedu.module.model.module.TrackedModule;
+import seedu.module.model.module.Link;
+
+/**
+ * Adds or launches a Link in a Module depending on input
+ */
+public class LinkCommand extends Command {
+    public static final String COMMAND_WORD = "link";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds or launches a Link to a Module. "
+            + "Parameters: " + "Index"
+            + "Action" + "t/Link title [l/Link]\n"
+            + "Example: " + COMMAND_WORD + "1 " + "a/add "
+            + "t/example " + "l/ " + "http://google.com\n"
+            + COMMAND_WORD + "1 "+ "a/go " + "t/" + "example\n";
+
+    public static final String MESSAGE_LINK_SUCCESS = "New link added to module.";
+    public static final String MESSAGE_LAUNCH_SUCCESS = "Website opened in default browser.";
+    public static final String MESSAGE_DUPLICATE_TITLE = "Link name already exists, choose another one.";
+
+    private final Index targetIndex;
+    private final Link webLink;
+    private final String action;
+    private final String linkTitle;
+
+    /**
+     * specific to `link add` commands
+     * @param action
+     * @param targetIndex
+     * @param webLink
+     */
+    public LinkCommand(String action, Index targetIndex, Link webLink) {
+        this.action = action;
+        this.targetIndex = targetIndex;
+        this.webLink = webLink;
+        this.linkTitle = null;
+    }
+
+    /**
+     * specific to `link go` commands
+     * @param action
+     * @param targetIndex
+     * @param linkTitle
+     */
+    public LinkCommand(String action, Index targetIndex, String linkTitle){
+        this.action = action;
+        this.targetIndex = targetIndex;
+        this.webLink = null;
+        this.linkTitle = linkTitle;
+    }
+
+    private Link find(TrackedModule targetModule, String linkTitle) throws CommandException{
+        for (Link l: targetModule.links){
+            if (l.name.equals(linkTitle)){
+                return l;
+            }
+        }
+        throw new CommandException("Link with matching title not found");
+    }
+    @Override
+    public CommandResult execute(Model model) throws CommandException{
```

Comment:
Same as mentioned above. Do include the execute method for find as well, and parse the find command with its own parser. Also if if-else statements, it might be better to include an else statement instead of just if and else if to catch unexpected outcomes.